agent/containermetadata/parse_metadata.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+
+	"github.com/cihub/seelog"
+	docker "github.com/fsouza/go-dockerclient"
+)
+
+// parseNetworkMetadata parses the docker.NetworkSettings struct and
+// packages the desired metadata for JSON marshaling
+// Since we accept incomplete metadata fields, we should not return
+// errors here and handle them at this stage.
+func parseNetworkMetadata(settings *docker.NetworkSettings) (NetworkMetadata, error) {
+	// If network settings are unavailable, users should be notified in case
+	// this was not intended
+	if settings == nil {
+		err := fmt.Errorf("parse network metadata: could not find network settings")
+		return NetworkMetadata{}, err
+	}
+
+	// This metadata is available in two different places in NetworkSettings
+	ipv4Address := settings.IPAddress
+
+	// Network mode is not available for Docker API versions 1.17-1.20
+	networkModeFromContainer := ""
+	if len(settings.Networks) == 1 {
+		for modeFromSettings, containerNetwork := range settings.Networks {
+			networkModeFromContainer = modeFromSettings
+			ipv4Address = containerNetwork.IPAddress
+		}
+	}
+
+	networkMD := NetworkMetadata{
+		networkMode: networkModeFromContainer,
+		ipv4Address: ipv4Address,
+	}
+	return networkMD, nil
+}
+
+// parseDockerContainerMetadata parses the metadata in a docker container
+// and packages this data for JSON marshaling
+// Since we accept incomplete metadata fields, we should not return
+// errors here and handle them at this stage.
+func parseDockerContainerMetadata(container *docker.Container) DockerContainerMD {
+	if container == nil {
```

Comment:
The reason it is consumed silently is the overall policy of continuing to parse the metadata even if some or all fields are missing. I am not sure if there is any reason to handle an error upstream if one is returned here because of the above policy means the error's only use is for logging.